pkg/controller/identity/iamgroupusermembership/controller.go

```diff
@@ -0,0 +1,202 @@
+	kube   client.Client
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMGroupUserMembership)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	observed, err := e.client.ListGroupsForUserRequest(&awsiam.ListGroupsForUserInput{
+		UserName: cr.Spec.ForProvider.UserName,
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(iam.IsErrorNotFound, err), errGet)
+	}
+
+	var attachedGroupObject *awsiam.Group
+	for i, group := range observed.Groups {
+		if cr.Spec.ForProvider.GroupName == group.GroupName {
+			attachedGroupObject = &observed.Groups[i]
+			break
+		}
+	}
+
+	if attachedGroupObject == nil {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	current := cr.Spec.ForProvider.DeepCopy()
+
+	// LateInitializeUserGroup
+	*cr.Spec.ForProvider.GroupName = awsclients.LateInitializeString(aws.StringValue(cr.Spec.ForProvider.GroupName), attachedGroupObject.GroupName)
+
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	cr.Status.AtProvider = v1alpha1.IAMGroupUserMembershipObservation{
+		AttachedGroupARN: aws.StringValue(attachedGroupObject.Arn),
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: true,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMGroupUserMembership)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Creating())
+
+	_, err := e.client.AddUserToGroupRequest(&awsiam.AddUserToGroupInput{
+		GroupName: cr.Spec.ForProvider.GroupName,
+		UserName:  cr.Spec.ForProvider.UserName,
+	}).Send(ctx)
+
+	return managed.ExternalCreation{}, errors.Wrap(err, errAdd)
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	// Updating any field will create a new Group-User Membership in AWS, which will be
+	// irrelevant/out-of-sync to the original defined attachment.
+	// It is encouraged to instead create a new IAMGroupUserMembership resource.
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
+	cr, ok := mgd.(*v1alpha1.IAMGroupUserMembership)
+	if !ok {
+		return errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Deleting())
+
+	_, err := e.client.RemoveUserFromGroupRequest(&awsiam.RemoveUserFromGroupInput{
+		GroupName: cr.Spec.ForProvider.GroupName,
+		UserName:  cr.Spec.ForProvider.UserName,
+	}).Send(ctx)
+
+	return errors.Wrap(resource.Ignore(iam.IsErrorNotFound, err), errRemove)
```

Comment:
I don't think AWS returns `NotFound` if the user is not member of the group already. Have you tested this manually? It's not critical since `Observe` handles the not attached case manually but if `iam.IsErrorNotFound` is non-functional, we may as well just remove it and not ignore any error.